File: lib/src/main/java/czh/fast/lib/application/BaseAPP.java
package czh.fast.lib.application;

import android.app.Application;

import com.facebook.cache.disk.DiskCacheConfig;
import com.facebook.common.memory.MemoryTrimType;
import com.facebook.common.memory.MemoryTrimmable;
import com.facebook.common.memory.MemoryTrimmableRegistry;
import com.facebook.common.memory.NoOpMemoryTrimmableRegistry;
import com.facebook.common.util.ByteConstants;
import com.facebook.drawee.backends.pipeline.Fresco;
import com.facebook.imagepipeline.core.ImagePipelineConfig;
import com.facebook.imagepipeline.core.ImagePipelineFactory;
import com.vise.log.ViseLog;
import com.vise.log.inner.LogcatTree;

abstract public class BaseAPP extends Application {

    @Override
    public void onCreate() {
        super.onCreate();
        initLog();
        initFresco();
    }

    private void initLog() {
        ViseLog.getLogConfig().configAllowLog(true)//是否输出日志
                .configShowBorders(false);//是否排版显示
        ViseLog.plant(new LogcatTree());//添加打印日志信息到Logcat的树
    }



    private void initFresco() {
        DiskCacheConfig diskCacheConfig = DiskCacheConfig.newBuilder(this)
                .setBaseDirectoryName("ImageCache").setMaxCacheSize(200 * ByteConstants.MB)
                .setMaxCacheSizeOnLowDiskSpace(100 * ByteConstants.MB)
                .setMaxCacheSizeOnVeryLowDiskSpace(50 * ByteConstants.MB)
                .setMaxCacheSize(80 * ByteConstants.MB).build();


        MemoryTrimmableRegistry memoryTrimmableRegistry = NoOpMemoryTrimmableRegistry
                .getInstance();
        memoryTrimmableRegistry.registerMemoryTrimmable(new MemoryTrimmable() {
            @Override
            public void trim(MemoryTrimType trimType) {
                final double suggestedTrimRatio = trimType.getSuggestedTrimRatio();

                if (MemoryTrimType.OnCloseToDalvikHeapLimit.getSuggestedTrimRatio() ==
                        suggestedTrimRatio
                        || MemoryTrimType.OnSystemLowMemoryWhileAppInBackground
                        .getSuggestedTrimRatio() == suggestedTrimRatio
                        || MemoryTrimType.OnSystemLowMemoryWhileAppInForeground
                        .getSuggestedTrimRatio() == suggestedTrimRatio) {
                    //清空内存缓存
                    ImagePipelineFactory.getInstance().getImagePipeline().clearMemoryCaches();
                }
            }
        });

        ImagePipelineConfig config = ImagePipelineConfig.newBuilder(this)
                .setMainDiskCacheConfig(diskCacheConfig)
                .setMemoryTrimmableRegistry(memoryTrimmableRegistry)
                .setDownsampleEnabled(true).build();
        Fresco.initialize(this, config);
    }

}

File: lib/src/main/java/czh/fast/lib/application/BaseApp.kt
package czh.fast.lib.application

import android.app.Application
import android.content.Context
import com.vise.log.ViseLog
import com.vise.log.inner.LogcatTree
import com.vise.utils.assist.SSLUtil
import com.vise.xsnow.http.ViseHttp
import com.vise.xsnow.http.interceptor.HttpLogInterceptor
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory
import java.util.*

/**
//┏┓　　　┏┓
//┏┛┻━━━┛┻┓
//┃　　　　　　　┃ 　
//┃　　　━　　　┃
//┃　┳┛　┗┳　┃
//┃　　　　　　　┃
//┃　　　┻　　　┃
//┃　　　　　　　┃
//┗━┓　　　┏━┛
//    ┃　　　┃   神兽保佑　　　　　　　　
//    ┃　　　┃   代码无BUG！
//    ┃　　　┗━━━┓
//    ┃　　　　　　　┣┓
//    ┃　　　　　　　┏┛
//    ┗┓┓┏━┳┓┏┛
//      ┃┫┫　┃┫┫
//      ┗┻┛　┗┻┛

 * Created by Dell on 2018/5/20
 * @author czh.
 */


abstract class BaseApp : Application() {
    abstract val BaseUrl: String

    override fun onCreate() {
        super.onCreate()
        initLog()
        initNet()
    }


    private fun initLog() {
        ViseLog.getLogConfig()
                .configAllowLog(true)//是否输出日志
                .configShowBorders(false)//是否排版显示
        ViseLog.plant(LogcatTree())//添加打印日志信息到Logcat的树
    }

    private fun initNet() {
        ViseHttp.init(this)
        ViseHttp.CONFIG()
                .baseUrl(BaseUrl)
                .globalHeaders(HashMap<String, String>())
                .globalParams(HashMap<String, String>())
                .readTimeout(30)
                .writeTimeout(30)
                .connectTimeout(30)
                .retryCount(3)
                .retryDelayMillis(6000)
                .setCookie(true)
                .interceptor(HttpLogInterceptor()
                        .setLevel(HttpLogInterceptor.Level.BODY))
                .converterFactory(GsonConverterFactory.create())
                .callAdapterFactory(RxJava2CallAdapterFactory.create())
                .SSLSocketFactory(SSLUtil.getSslSocketFactory(null, null, null))

    }

}